nbclient/client.py

```diff
@@ -642,26 +688,22 @@ async def async_wait_for_reply(self, msg_id, cell=None):
             else:
                 if msg['parent_header'].get('msg_id') == msg_id:
                     return msg
+        return None
 
     wait_for_reply = run_sync(async_wait_for_reply)
     # Backwards compatability naming for papermill
     _wait_for_reply = wait_for_reply
 
-    def _timeout_with_deadline(self, timeout, deadline):
-        if deadline is not None and deadline - monotonic() < timeout:
-            timeout = deadline - monotonic()
-
-        if timeout < 0:
-            timeout = 0
-
-        return timeout
```

Comment:
Note that this method seemed to be never called, so I removed it.

Reply:
Huh, wonder how that happened. Probably we missed nuking it in an earlier refactor. Good catch.